### PR TITLE
Improvement: Minigame call order, fixed active state, added possibility to modify the EPOP

### PR DESCRIPTION
**Pull request by Alf21**
- Added `MINIGAME:ShowActivationEPOP()` as a possibility to overwrite the EPOP with an own version
- Fixed `active` state
- Improved the `minigame`'s call order
  - Added `MINIGAME:PreInitialize()`
  - Added `MINIGAME:SetupData()`
  - Added `MINIGAME:Initialize()`
- Fixed EPOP bug reasoned by #5 
- Fixed incorrect ConVar replication (ULX) for `ttt2_minigames_autostart_rounds` and `rep_ttt2_minigames_autostart_random` ConVar
- Expanded the size of basic settings to allow for all ConVars to be visible
- Simplified a few labels to fix graphical bugs
- Added minigame enabled and minigame random chance ConVars to ULX menu
- Fixed wrong ConVar replication (ULX) order
- Added `TTT2MinigamesLoaded` hook

### CODE REVIEW

File: lua/minigames/minigames/minigame_base/shared.lua [hunk 1]
```diff
@@ -58,4 +65,20 @@ if SERVER then
...
+	function MINIGAME:ShowActivationEPOP()
+		if not istable(self.lang) or not EPOP then return end
+
+		EPOP:AddMessage({
+				text = LANG.TryTranslation("ttt2_minigames_" .. self.name .. "_name"),
+				color = COLOR_ORANGE
+			},
+			self.lang.desc and LANG.TryTranslation("ttt2_minigames_" .. self.name .. "_desc") or nil,
+			12,
+			true
```
Comment (ZacharyHinds) [inline lua/minigames/minigames/minigame_base/shared.lua:102]: What purpose does this `true` serve?

When I tested this, I was getting the following error:

[TTT2 (Base) - v0.7.3b - New Roles, Extensible, Better Performance, ULX Support] gamemodes/terrortown/gamemode/shared/hud_elements/ttteventpopup/pure_skin_eventpopup.lua:102: attempt to get length of field 'iconTable' (a boolean value)
  1. PrepareItem - gamemodes/terrortown/gamemode/shared/hud_elements/ttteventpopup/pure_skin_eventpopup.lua:102
   2. Draw - gamemodes/terrortown/gamemode/shared/hud_elements/ttteventpopup/pure_skin_eventpopup.lua:122
    3. DrawElemAndChildren - gamemodes/terrortown/gamemode/shared/huds/base_huds/hud_base/cl_init.lua:304
     4. Draw - gamemodes/terrortown/gamemode/shared/huds/base_huds/hud_base/cl_init.lua:330
      5. DrawHUD - gamemodes/terrortown/gamemode/client/cl_hud_manager.lua:57
       6. unknown - gamemodes/terrortown/gamemode/client/cl_hud_manager.lua:97

But if I remove this `true` it no longer gives an error.

Comment (Alf21) [reply lua/minigames/minigames/minigame_base/shared.lua:102]: @Histalek https://github.com/TTT-2/ttt2mg/pull/5

Comment (Histalek) [reply lua/minigames/minigames/minigame_base/shared.lua:102]: > What purpose does this `true` serve?
> 
> When I tested this, I was getting the following error:
> 
> [TTT2 (Base) - v0.7.3b - New Roles, Extensible, Better Performance, ULX Support] gamemodes/terrortown/gamemode/shared/hud_elements/ttteventpopup/pure_skin_eventpopup.lua:102: attempt to get length of field 'iconTable' (a boolean value)
> 
>     1. PrepareItem - gamemodes/terrortown/gamemode/shared/hud_elements/ttteventpopup/pure_skin_eventpopup.lua:102
> 
>     2. Draw - gamemodes/terrortown/gamemode/shared/hud_elements/ttteventpopup/pure_skin_eventpopup.lua:122
>        3. DrawElemAndChildren - gamemodes/terrortown/gamemode/shared/huds/base_huds/hud_base/cl_init.lua:304
>        4. Draw - gamemodes/terrortown/gamemode/shared/huds/base_huds/hud_base/cl_init.lua:330
>        5. DrawHUD - gamemodes/terrortown/gamemode/client/cl_hud_manager.lua:57
>        6. unknown - gamemodes/terrortown/gamemode/client/cl_hud_manager.lua:97
> 
> 
> But if I remove this `true` it no longer gives an error.

The true makes the epop message blocking, meaning it stays on screen for the full duration and can't be overwritten by another epop message. 

there seems to be missing an argument between the 12 and the true, `nil` should work i think as the icontable is optional

File: lua/minigames/minigames/minigame_base/shared.lua [hunk 2]
```diff
@@ -58,4 +65,20 @@ if SERVER then
...
+		)
```
Comment (Histalek) [inline lua/minigames/minigames/minigame_base/shared.lua]:
```suggestion
		EPOP:AddMessage({
				text = LANG.TryTranslation("ttt2_minigames_" .. self.name .. "_name"),
				color = COLOR_ORANGE
			},
			self.lang.desc and LANG.TryTranslation("ttt2_minigames_" .. self.name .. "_desc") or nil,
			12,
			nil,
			true
		)
```
this should work (untested)
also i don't know why i didn't stumble upon this error when i did #5 ...

**Comment by Alf21** [comment]
@ZacharyHinds  I will merge it as soon as you have approved/said, those changes are okay because it seems you using this base the most ^^

**Review by ZacharyHinds — APPROVED** [review]

**Pull request merged**